### PR TITLE
increase cassandra pool size to 30->100 from 20->100 connections

### PR DESCRIPTION
**Pull request by jnelson15**
Followup to https://github.com/palantir/atlasdb/pull/1336

@jboreiko and I discussed and decided we'll bump default here and then change Atlas config in our large internal product to reflect old config to avoid screwing with anyone with overriden defaults.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1402)

### CODE REVIEW

**Comment by tpetracca** [comment]
I'd argue that the default min and max be even closer together. Is there a reason for wanting the ability to fluctuate to such a degree?

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -65,6 +65,10 @@ develop
          - Enable garbage collection logging for CircleCI builds.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1398>`__)
 
+    *    - |improved|
+         - Increase minimum Cassandra pool size to better handle bursts in requests.
```
Comment (schlosna) [inline docs/source/release_notes/release-notes.rst]: should indicate that these are the defaults that were changed and the new values

**Comment by jnelson15** [comment]
@tpetracca what do you think is a more appropriate range? I think the original intention was to minimize resources used since this per-node while still allowing it to increase when under heavy usage.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java [hunk 2]
```diff
@@ -208,7 +208,7 @@ protected final void check() {
         }
         Preconditions.checkNotNull(keyspace(), "'keyspace' must be specified");
         double evictionCheckProportion = proportionConnectionsToCheckPerEvictionRun();
-        Preconditions.checkArgument(evictionCheckProportion > 0 && evictionCheckProportion <= 1,
+        Preconditions.checkArgument(evictionCheckProportion > 0.01 && evictionCheckProportion <= 1,
                 "'proportionConnectionsToCheckPerEvictionRun' must be between 0 and 1");
```
Comment (hsaraogi) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java]: Change the error message as well "'proportionConnectionsToCheckPerEvictionRun' must be between 0.01 and 1");".

**Comment by tpetracca** [comment]
So I'm in favor of merging this as a strict improvement and if we continue to see weirdness around the burstability causing too much overhead then we can change

**Pull request merged**